tests/ui/blocks_in_if_conditions.rs

```diff
@@ -61,4 +61,16 @@ fn block_in_assert() {
     );
 }
 
+// issue #11814
+fn block_in_match_expr(num: i32) -> i32 {
+    match {
```

Comment:
I'd like to see a test with an unsafe block, too (just so be sure it's handled correctly).